### PR TITLE
 fix: check for null [#560] && fix lint errors

### PR DESCRIPTION
**Pull request by thaddeusreid**
**What kind of change does this PR introduce?**

- [x] bugfix
- [ ] feature
- [ ] refactoring / style
- [ ] build / chore
- [ ] documentation

**Did you add tests for your changes?**

- [ ] Yes, my code is well tested
- [x] Not relevant

**If relevant, did you update the documentation?**

- [ ] Yes, I've updated the documentation
- [x] Not relevant

**Summary**

Per [#560] a TypeError is thrown because `this.fileInputEl.files` is referenced in cases where `this.fileInputEl` is null.

I also fixed 3 eslint errors that address comment indenting/whitespace.

**Does this PR introduce a breaking change?**

no

**Other information**

### CODE REVIEW

**Comment by thaddeusreid** [comment]
Hmm, looks like TravisCI is failing on a lint error that the `"precommit": "lint-staged"` will correct even when I am intentionally committing the lint changes TravisCI wants to see.

**Comment by okonet** [comment]
I’m traveling now but I’ll check what’s wrong here next week and get back to you.

**Comment by thaddeusreid** [comment]
Strange. The eslint phase of the TravisCI build asks for one thing:
```
/home/travis/build/react-dropzone/react-dropzone/src/index.js
  401:10  error  Replace `⏎··········...divProps·/*·expand·user·provided·props·first·so·event·handlers·are·never·overridden·*/⏎········` with `...divProps·/*·expand·user·provided·props·first·so·event·handlers·are·never·overridden·*/`        prettier/prettier
  415:12  error  Replace `⏎············...inputProps·/*·expand·user·provided·inputProps·first·so·inputAttributes·override·them·*/⏎··········` with `...inputProps·/*·expand·user·provided·inputProps·first·so·inputAttributes·override·them·*/`  prettier/prettier
```

Yet when I manually change my local file to match what Travis is expecting, the `precommit` script in package.json applies an `eslint --fix` to undo my change. Seems like there must be a version mismatch between my local and TravisCI.

Here are my local versions of eslint and lint-staged:
```bash
$ ./node_modules/.bin/eslint --version
v4.16.0
$ ./node_modules/.bin/lint-staged --version
4.3.0
```

**Comment by okonet** [comment]
Weird. Travis uses the versions from `package.json` so I doubt there is a version mismatch. Must be something else.

**Comment by okonet** [comment]
Have you tried running `eslint --fix` locally?

**Comment by thaddeusreid** [comment]
@okonet yes.

`eslint --fix`  it styles the code like this:
```javascript
{
  ...divProps /* expand user provided props first so event handlers are never overridden */
 }
```
while TravisCI wants this:
```javascript
{...divProps /* expand user provided props first so event handlers are never overridden */}
```
This is so dirty... I removed `"precommit": "lint-staged",` from my local package.json so that I could commit the lint style TravisCI wanted... and now it's passing.

**Comment by rkmarks** [comment]
To be honest @thaddeusreid I did the same for my contribution. It might have something to do w/ the way lint-staged is setup. I'll look into that. nice work though! 👍

File: src/index.js [hunk 1]
```diff
@@ -239,11 +239,13 @@ class Dropzone extends React.Component {
...
       setTimeout(() => {
-        // Returns an object as FileList
-        const { files } = this.fileInputEl
+        if (this.fileInputEl != null) {
+          // Returns an object as FileList
+          const { files } = this.fileInputEl
 
-        if (!files.length) {
-          this.isFileDialogActive = false
+          if (!files.length) {
```
Comment (yuri-sakharov) [inline src/index.js]: simpler to add `if (!files || !files.length)` add remove this `if (this.fileInputEl != null) {`

Comment (okonet) [reply src/index.js]: I don't know. The PR code looks more readable to me

Comment (yuri-sakharov) [reply src/index.js]: for me more readable this version 
```
if (this.isFileDialogActive) {
                setTimeout(() => {
                    const { files } = this.fileInputEl;

                    if (!files || !files.length) {
                        this.isFileDialogActive = false;
                    }
                    if (typeof onFileDialogCancel === 'function') {
                        onFileDialogCancel();
                    }
                }, 300);
            }
```

Comment (yuri-sakharov) [reply src/index.js]: no need one more if and what about case with `files undefined`

**Comment by okonet** [comment]
@rkmarks @thaddeusreid I just checked out the latest master and tried to commit some changes. `eslint --fix` did format it correctly (the way Travis CI would expect) so I'm not sure what's going on.

Please provide me with more details on how did you format the code so it was passing local checks but not the CI. One thing that can cause this is not latest dependencies being used. Please make sure you run `yarn install` before running `eslint --fix .` and `lint-staged`.

**Comment by thaddeusreid** [comment]
@okonet Ahh, that's the problem. I installed dependencies with `npm install`.

I can see the same results as TravisCI after I removing node_modules, `yarn install`, and `yarn run eslint:src`.

Just a habit of mine to use `npm` over `yarn`. Thanks for the merge 👍

**Pull request merged**